File: languages/ruby/05-testing/src/calculator.rb
# A simple calculator class to test
class Calculator
  def add(a, b)
    a + b
  end

  def subtract(a, b)
    a - b
  end

  def multiply(a, b)
    a * b
  end

  def divide(a, b)
    raise ArgumentError, "Cannot divide by zero" if b.zero?
    a / b
  end

  def power(base, exp)
    raise ArgumentError, "Negative exponent not supported" if exp < 0
    result = 1
    exp.times { result *= base }
    result
  end
end

File: languages/javascript/07-modules-and-packages/exercises/calculator.js
// calculator.js - Default export module
export default function calculate(a, op, b) {
  switch (op) {
    case '+':
      return a + b;
    case '-':
      return a - b;
    case '*':
      return a * b;
    case '/':
      return b !== 0 ? a / b : 'Cannot divide by zero';
    default:
      return 'Unknown operator';
  }
}
